trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java

```diff
@@ -0,0 +1,130 @@
+     */
+    public FunctionalTestWriter(@NonNull final OutputStream outputStream, @NonNull final PrintColor printColor) {
+        super(outputStream);
+        this.coloringCode = createColorCode(printColor);
+        this.extraLength = indentation.length + coloringCode.length + escapeChar.length;
+    }
+
+    /**
+     * Inserts to the start the escape char.
+     *
+     * @param chars the given chars to escape.
+     * @return the given chars with the escape char.
+     */
+    @NonNull
+    private static char[] concatEscapeChar(@NonNull final char[] chars) {
+        return concatCharArrays(escapeChar, chars);
+    }
+
+    /**
+     * Concatenates two arrays of chars.
+     *
+     * @param first  the first array.
+     * @param second the second array.
+     * @return the concatenated result.
+     */
+    @NonNull
+    private static char[] concatCharArrays(@NonNull final char[] first, @NonNull final char[] second) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(first);
+        stringBuilder.append(second);
+        return stringBuilder.toString().toCharArray();
+    }
+
+    /**
+     * Creates a color code from the given {@link PrintColor}.
+     *
+     * @param printColor the given PrintColor.
+     * @return the char array representation of the given color.
+     */
+    @NonNull
+    private static char[] createColorCode(@NonNull final PrintColor printColor) {
+        return ("[" + printColor.getColorCode() + "m").toCharArray();
+    }
+
+    /**
+     * Resets the console to the default color.
+     */
+    public static void reset() {
+        System.out.println(concatEscapeChar(createColorCode(PrintColor.DEFAULT)));
+    }
+
+    @Override
+    public void write(@NonNull final char[] chars, final int i, final int i1) {
+        final char[] coloredChars = color(chars);
+        final char[] output = addIndent(coloredChars);
+        super.write(output, i, i1 + extraLength);
+    }
+
+    /**
+     * Inserts to the start of the given char array the {@link #indentation}.
+     *
+     * @param chars the given chars.
+     * @return the indented array of chars.
+     */
+    @NonNull
+    private char[] addIndent(@NonNull final char[] chars) {
+        return concatCharArrays(indentation, chars);
+    }
+
+    /**
+     * Inserts to the start the coloring chars to a given array of chars.
+     *
+     * @param chars the given array of chars.
+     * @return the given text with the coloring chars.
+     */
+    @NonNull
+    private char[] color(@NonNull final char[] chars) {
+        final char[] escaped = concatEscapeChar(coloringCode);
+        return concatCharArrays(escaped, chars);
+    }
+
+    /**
+     * Enum class for different colors. This is based on the ANSI escape codes-
+     *
+     * @see <a href="https://en.wikipedia.org/wiki/ANSI_escape_code">https://en.wikipedia.org/wiki/ANSI_escape_code</a>
+     */
+    public enum PrintColor {
```

Comment:
Note: only 3 of them are used, but I thought we could leave the rest of them here too

Reply:
Moving [this conversation](https://github.com/bitrise-io/trace-android-sdk/pull/194#issuecomment-831190219) here too as they're all similar.

How much effort would it be to get the bitrise colours here from [bitkit](https://bitkit.netlify.app/documentation/materials/colors) - perhaps using red4 and aqua4 as the two colours? 

i think if we did go with the bitrise colours, it would make sense to remove the other colours we have (maybe except black and white?) - and link to bitkit to let people know where they came from? What do you think?

Reply:
We can use the bitkit colors too. Updated code.
![Screenshot 2021-05-04 at 13 46 20](https://user-images.githubusercontent.com/51914448/116998853-2138ff00-acdf-11eb-812a-0266ca78d241.png)

Reply:
awesome 🎉 this would make a good 5 minute tech improvement talk btw if you're interested :)